lizmap/modules/lizmap/lib/Request/WFSRequest.php

```diff
@@ -346,6 +348,115 @@ protected function process_getfeature()
      */
     protected function getfeatureQgis()
     {
+        // In the WFS OGC standard FEATUREID and BBOX parameters cannot be mutually set
+        // but in Lizmap, the user can do a selection, based on featureid, and can request
+        // a download, a WFS GetFeature request, based on this selection with a restriction
+        // to map extent, sofeatureid and bbox parameter can be set mutually and featureid
+        // parameter needs to be transform in an expression filter.
+        // The transformation is only available if the QGIS layer has been set.
+        if ($this->param('featureid')
+            && $this->param('bbox')
+            && $this->qgisLayer) {
+            $provider = $this->qgisLayer->getProvider();
+
+            $typename = $this->requestedTypename();
+            $featureid = $this->param('featureid', '');
+
```

Comment:
I suggest to create a new method, `$expFilter = $this->getExpFilter($featureId, $typename)`, so getfeatureQgis will be more readable and not too long

Reply:
(and you could unit test this method)